### PR TITLE
Suggested documentation improvements

### PR DESCRIPTION
**Pull request by andrie**
Added:

* list of functions and descriptions in the package documentation
* improvemens in function documentation, particularly in `default_backend()`

### CODE REVIEW

File: R/default_backend.R [hunk 1]
```diff
@@ -2,32 +2,35 @@
 #' Select the default backend and default keyring
 #'
 #' The default backend is selected
-#' 1. based on the `keyring_backend` option. See [base::options()].
-#'    This can be set to a character string, and then the
-#'    *backend_*`string` class is used to create the default backend.
+#' 1. based on the `keyring_backend` option. See [options()][base::options].
```
Comment (gaborcsardi) [inline R/default_backend.R]: I think this is fine as `[base::options()]`.

File: R/default_backend.R [hunk 2]
```diff
@@ -2,32 +2,35 @@
...
 #' 1. If this is not set, then the `R_KEYRING_BACKEND` environment variable
 #'    is checked.
 #' 1. If this is not set, either, then the backend is selected
 #'    automatically, based on the OS:
 #'    1. On Windows, the Windows Credential Store (`"wincred"`) is used.
 #'    1. On macOS, Keychain services are selected (`"macos"`).
-#'    1. Linux uses the Secret Service API (`"secret_service"`),
-#'       and it also checks that the service is available. It is typically
-#'       only available on systems with a GUI.
+#'    1. Linux uses the Secret Service API (`"secret_service"`).
```
Comment (gaborcsardi) [inline R/default_backend.R]: I would keep the extra info.

File: R/default_backend.R [hunk 3]
```diff
@@ -2,32 +2,35 @@
...
+#'     - Change this value by using `Sys.setenv(R_KEYRING_KEYRING = "NEWVALUE")`, either in your script or in your `.Renviron` file. See [startup][base::startup] for information about using `.Renviron`
+#' 1. Finally, if neither of these are set, then the OS default keyring is selected.
+#'    - Usually this keyring is automatically unlocked when the user logs in.
 #'
 #' @param keyring Character string, the name of the keyring to use,
 #'   or `NULL` for the default keyring.
 #' @return The backend object itself.
+#' 
+#' 
+#' @seealso [backend_env()], [backend_macos()], [backend_secret_service()], [backend_wincred()]
```
Comment (gaborcsardi) [inline R/default_backend.R]: These are not functions, so just `[backend_env]`, `[backend_macos]`, etc.

File: R/package.R [hunk 4]
```diff
@@ -2,10 +2,38 @@
...
 #' Platform independent API to many system credential store
-#' implementations. Currently supported: Keychain on 'macOS', Credential
-#' Store on 'Windows', the Secret Service API and the Gnome keyring API on
-#' 'Linux'.
+#' implementations. Currently supported: 
+#' * Keychain on 'macOS'
+#' * Credential Store on 'Windows'
+#' * the Secret Service API and the Gnome keyring API on 'Linux'.
+#' 
+#' **Managing keyrings:**
```
Comment (gaborcsardi) [inline R/package.R]: This can be 

```
@section Managing keyrings:
```

File: R/package.R [hunk 5]
```diff
@@ -2,10 +2,38 @@
...
+#' 
+#' A keyring is a collection of keys that can be treated as a unit. A keyring typically has a name and a password to unlock it.
+#'     
+#' - [keyring_create()]
+#' - [keyring_delete()]
+#' - [keyring_list()]
+#' - [keyring_lock()]
+#' - [keyring_unlock()]
+#'   
+#' **Query secret keys in a keyring:**
```
Comment (gaborcsardi) [inline R/package.R]: This can be `@section` as well imo.

File: R/package.R [hunk 6]
```diff
@@ -2,10 +2,38 @@
...
+#' 
+#' Each keyring can contain one or many secrets (keys). A key is defined by a service name and a password. Once a key is defined, it persists in the keyring store of the operating system. This means the keys persist beyond the termination of and R session. Specifically, you can define a key once, and then read the key value in completely independent R sessions.
```
Comment (gaborcsardi) [inline R/package.R]: Can you pls break the lines around 75 characters? Thanks.

File: R/package.R [hunk 7]
```diff
@@ -2,10 +2,38 @@
...
+#' 
+#' - Setting a secret interactively: [key_set()]
+#' - Setting a secret from the console or script, i.e. non-interactively: [key_set_with_value()]
+#' - Reading a secret: [key_get()]
+#' - Listing secrets: [key_list()]
+#' - Deleting a secret: [key_delete()]
 #'
+#' **Configuring an OS-specific backend:**
```
Comment (gaborcsardi) [inline R/package.R]: `@section`

File: R/package.R [hunk 8]
```diff
@@ -2,10 +2,38 @@
...
+#'   - The default is operating system specific, and is described in [default_backend()]. In most cases you don't have to configure this.
+#'   - MacOS: [backend_macos()]
+#'   - Linux: [backend_secret_service()]
+#'   - Windows: [backend_wincred()]
+#'   - Or store the secrets in environment variables on other operating systems: [backend_env()]
```
Comment (gaborcsardi) [inline R/package.R]: No need for the `()` (x4).

File: R/package.R [hunk 9]
```diff
@@ -2,10 +2,38 @@
...
-#' Store on 'Windows', the Secret Service API and the Gnome keyring API on
-#' 'Linux'.
+#' implementations. Currently supported: 
+#' * Keychain on 'macOS'
+#' * Credential Store on 'Windows'
+#' * the Secret Service API and the Gnome keyring API on 'Linux'.
+#' 
+#' **Managing keyrings:**
+#' 
+#' A keyring is a collection of keys that can be treated as a unit. A keyring typically has a name and a password to unlock it.
```
Comment (gaborcsardi) [inline R/package.R]: Can you pls break lines at 75 characters?

File: R/default_backend.R [hunk 10]
```diff
@@ -2,32 +2,35 @@
...
 #'
 #' Most backends support multiple keyrings. For these the keyring is
 #' selected from
 #' 1. the supplied `keyring` argument (if not `NULL`), or
 #' 1. the `keyring_keyring` option.
+#'     - You can change this by using `options(keyring_keyring = "NEWVALUE")`
 #' 1. If this is not set, the `R_KEYRING_KEYRING` environment variable.
-#' 1. If this is not set, then the OS default keyring is selected.
-#'    Usually this keyring is automatically unlocked when the user logs in.
+#'     - Change this value by using `Sys.setenv(R_KEYRING_KEYRING = "NEWVALUE")`, either in your script or in your `.Renviron` file. See [startup][base::startup] for information about using `.Renviron`
```
Comment (gaborcsardi) [inline R/default_backend.R]: Can you pls break lines at 75 characters? Thanks.

**Comment by gaborcsardi** [comment]
Thanks, I added a bunch of small comments. Can you pls take a look at them?

File: R/default_backend.R (hunk 1 above)
Comment (andrie) [inline R/default_backend.R]: Done

File: R/default_backend.R (hunk 2 above)
Comment (andrie) [inline R/default_backend.R]: Fixing this. I think this slipped through when fixing a merge conflict.

File: R/default_backend.R (hunk 10 above)
Comment (andrie) [inline R/default_backend.R]: Done

File: R/default_backend.R (hunk 3 above)
Comment (andrie) [inline R/default_backend.R]: Good catch

File: R/package.R (hunk 4 above)
Comment (andrie) [inline R/package.R]: Agreed.

File: R/package.R (hunk 5 above)
Comment (andrie) [inline R/package.R]: Yes

File: R/package.R (hunk 6 above)
Comment (andrie) [inline R/package.R]: Done

File: R/package.R (hunk 8 above)
Comment (andrie) [inline R/package.R]: Yes

File: R/package.R (hunk 7 above)
Comment (andrie) [inline R/package.R]: Done

**Comment by gaborcsardi** [comment]
Great, thanks! I'll this docs typo: https://travis-ci.org/gaborcsardi/keyring/jobs/208671719#L757

**Comment by gaborcsardi** [comment]
I also used this to create a better README, so thanks again!

**Pull request merged**